// react-bank-card/es/index.js
var _templateObject = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n"], ["\n  width: 100%;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n"]),
    _templateObject2 = _taggedTemplateLiteralLoose(["\n  font-size: .875rem;\n  margin-left: .625rem;\n  width: 2rem;\n  text-align: left;\n"], ["\n  font-size: .875rem;\n  margin-left: .625rem;\n  width: 2rem;\n  text-align: left;\n"]),
    _templateObject3 = _taggedTemplateLiteralLoose(["font-size: 1.25rem;"], ["font-size: 1.25rem;"]),
    _templateObject4 = _taggedTemplateLiteralLoose(["\n  font-size: 1.5rem;\n  width: 100%;\n"], ["\n  font-size: 1.5rem;\n  width: 100%;\n"]),
    _templateObject5 = _taggedTemplateLiteralLoose(["margin-left: 10px;"], ["margin-left: 10px;"]),
    _templateObject6 = _taggedTemplateLiteralLoose(["\n  width: 330px;\n  transform-style: preserve-3d;\n  transition: all 400ms linear;\n  position: relative;\n\n  &::before {\n    content: '';\n    display: block;\n    padding-top: 66.66%; /* 3:2 ratio */\n  }\n\n  ", ";\n"], ["\n  width: 330px;\n  transform-style: preserve-3d;\n  transition: all 400ms linear;\n  position: relative;\n\n  &::before {\n    content: '';\n    display: block;\n    padding-top: 66.66%; /* 3:2 ratio */\n  }\n\n  ", ";\n"]),
    _templateObject7 = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 0;\n  left: 0;\n  width: 100%;\n  height: 100%;\n  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);\n  border-radius: 10px;\n  backface-visibility: hidden;\n  transform-style: preserve-3d;\n  display: flex;\n  flex-direction: column;\n  justify-content: space-between;\n"], ["\n  position: absolute;\n  top: 0;\n  left: 0;\n  width: 100%;\n  height: 100%;\n  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);\n  border-radius: 10px;\n  backface-visibility: hidden;\n  transform-style: preserve-3d;\n  display: flex;\n  flex-direction: column;\n  justify-content: space-between;\n"]),
    _templateObject8 = _taggedTemplateLiteralLoose(["\n  background-image: linear-gradient(to bottom right, #2f638b 50%, #3c7daf 50%);\n  padding: 25px;\n"], ["\n  background-image: linear-gradient(to bottom right, #2f638b 50%, #3c7daf 50%);\n  padding: 25px;\n"]),
    _templateObject9 = _taggedTemplateLiteralLoose(["\n  background-image: linear-gradient(to bottom left, #2f638b 50%, #3c7daf 50%);\n  padding: 5% 0;\n  transform: rotateY(180deg);\n\n  ", ":nth-child(2),\n  ", ":nth-child(3) {\n    padding: 0 .625rem;\n  }\n"], ["\n  background-image: linear-gradient(to bottom left, #2f638b 50%, #3c7daf 50%);\n  padding: 5% 0;\n  transform: rotateY(180deg);\n\n  ", ":nth-child(2),\n  ", ":nth-child(3) {\n    padding: 0 .625rem;\n  }\n"]),
    _templateObject10 = _taggedTemplateLiteralLoose([""], [""]),
    _templateObject11 = _taggedTemplateLiteralLoose(["\n  background: #000;\n  height: 3.125rem;\n  width: 100%;\n"], ["\n  background: #000;\n  height: 3.125rem;\n  width: 100%;\n"]),
    _templateObject12 = _taggedTemplateLiteralLoose(["\n  flex: 1 1 auto;\n  background: #fff;\n  height: 2.5rem;\n"], ["\n  flex: 1 1 auto;\n  background: #fff;\n  height: 2.5rem;\n"]);

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteralLoose(strings, raw) { strings.raw = raw; return strings; }

import PropTypes from "prop-types";
import React, { Component } from "react";
import styled from "styled-components";

import Chip from "./components/Chip";
import Field from "./components/Field";
import Logo from "./components/Logo";

var DEFAULT_NAME = "•••••• •••••••";
var DEFAULT_NUMBER = "••••••••••••••••";
var DEFAULT_EXPIRY_MONTH = "••";
var DEFAULT_EXPIRY_YEAR = "••";
var DEFAULT_CVC = "•••";

var Line = styled.div(_templateObject);

var CvcField = Field.extend(_templateObject2);

var NameField = Field.extend(_templateObject3);

var NumberField = Field.extend(_templateObject4);

var Expiry = styled.div(_templateObject5);

var Card = styled.div(_templateObject6, function (props) {
  return props.flipped && "transform: rotateY(180deg);";
});

var Face = styled.div(_templateObject7);

var Recto = Face.extend(_templateObject8);

var Verso = Face.extend(_templateObject9, Line, Line);

var VersoChip = Chip.extend(_templateObject10);

var Strip = styled.div(_templateObject11);
var Signature = styled.div(_templateObject12);

var ReactBankCard = function (_Component) {
  _inherits(ReactBankCard, _Component);

  function ReactBankCard() {
    _classCallCheck(this, ReactBankCard);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  ReactBankCard.prototype.renderRecto = function renderRecto() {
    var _props = this.props,
        expiryMonth = _props.expiryMonth,
        expiryYear = _props.expiryYear,
        highlighted = _props.highlighted,
        name = _props.name,
        number = _props.number;

    var trimedNumber = number.replace(/[^\d]+/, "") || DEFAULT_NUMBER;
    var spacedNumber = trimedNumber.replace(/(.{4})/g, "$1 ");
    return React.createElement(
      Recto,
      null,
      React.createElement(
        Line,
        null,
        React.createElement(Chip, null),
        React.createElement(Logo, { number: number })
      ),
      React.createElement(
        Line,
        null,
        React.createElement(
          NumberField,
          { highlighted: highlighted && highlighted.number },
          spacedNumber
        )
      ),
      React.createElement(
        Line,
        null,
        React.createElement(
          NameField,
          { highlighted: highlighted && highlighted.name },
          name || DEFAULT_NAME
        ),
        React.createElement(
          Expiry,
          null,
          React.createElement(
            Field,
            { highlighted: highlighted && highlighted.expiryMonth },
            expiryMonth || DEFAULT_EXPIRY_MONTH
          ),
          React.createElement(
            Field,
            { highlighted: highlighted && highlighted.expiryYear },
            expiryYear || DEFAULT_EXPIRY_YEAR
          )
        )
      )
    );
  };

  ReactBankCard.prototype.renderVerso = function renderVerso() {
    var _props2 = this.props,
        cvc = _props2.cvc,
        highlighted = _props2.highlighted;

    return React.createElement(
      Verso,
      null,
      React.createElement(
        Line,
        null,
        React.createElement(Strip, null)
      ),
      React.createElement(
        Line,
        null,
        React.createElement(Signature, null),
        React.createElement(
          CvcField,
          { highlighted: highlighted && highlighted.cvc },
          cvc || DEFAULT_CVC
        )
      ),
      React.createElement(
        Line,
        null,
        React.createElement(VersoChip, null)
      )
    );
  };

  ReactBankCard.prototype.render = function render() {
    var verso = this.props.verso;

    return React.createElement(
      Card,
      { flipped: verso },
      this.renderRecto(),
      this.renderVerso()
    );
  };

  return ReactBankCard;
}(Component);

ReactBankCard.propTypes = process.env.NODE_ENV !== "production" ? {
  cvc: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  expiryMonth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  expiryYear: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  highlighted: PropTypes.object,
  name: PropTypes.string,
  number: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  verso: PropTypes.bool
} : {};

ReactBankCard.defaultProps = {
  cvc: DEFAULT_CVC,
  expiryMonth: DEFAULT_EXPIRY_MONTH,
  expiryYear: DEFAULT_EXPIRY_YEAR,
  highlighted: null,
  name: DEFAULT_NAME,
  number: DEFAULT_NUMBER,
  verso: false
};

export default ReactBankCard;
